### ISSUE TITLE
Documenter web page?

### ISSUE THREAD

**Issue by gdalle** [SciML/ADTypes.jl#25]
**Is your feature request related to a problem? Please describe.**

Since #23 we have docstrings, but they are not shown anywhere right now. As a result, the API specification is still imprecise for users, which is a problem when so much of SciML depends on this package.

**Describe the solution you’d like**

A Documenter.jl web page showcasing the docstrings and the list of exported names.

**Describe alternatives you’ve considered**

I tried adding a list of exports to the README in #23, but it will become periodically out of date (it already is since #21 was merged too). A Documenter workflow is more adaptive.

**Additional context**

I can set up the Documenter stuff by myself, but there are things I cannot do:

- Add the required secrets to the repo. If you want to give me access that's also fine.
- Integrate this to a wider SciML multidoc. Not sure we want to do that though, because I envision this package being foundational outside of SciML too thanks to https://github.com/gdalle/DifferentiationInterface.jl

**Comment by Vaibhavdixit02** [comment]
I can set it up. But in case you had already done something for this I have added you to the repository.

**Comment by gdalle** [comment]
Thanks! I can do it as well, once #26 is merged

Once there is a nice page up and running we can bump the patch version and release, if that's okay?
